lldb/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp

```diff
@@ -1858,6 +1862,57 @@ bool ScriptInterpreterPythonImpl::GetScriptedSummary(
   return ret_val;
 }
 
+Status ScriptInterpreterPythonImpl::RecognizeType(
+    const char *p_function_name, const lldb::ValueObjectSP input_valobj,
+    CompilerType &output_ct, Address &output_addr) {
+  if (!p_function_name || p_function_name[0] == '\0' || !input_valobj.get())
+    return Status("No python function name");
+
+  Locker py_lock(this,
+                 Locker::AcquireLock | Locker::InitSession | Locker::NoSTDIN);
+
+  PythonObject ret_val = SWIGBridge::LLDBSwigPythonCallRecognizerScript(
```

Comment:
We should discuss error handling between this function and `LLDBSwigPythonCallRecognizerScript` on a call